custom_components/tahoma/coordinator.py

```diff
@@ -0,0 +1,60 @@
+"""Helpers to help coordinate updates."""
+from datetime import timedelta
+import logging
+from typing import Dict, List, Optional, Union
+
+from tahoma_api.client import TahomaClient
+from tahoma_api.models import DataType, Device, State
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+TYPES = {
+    DataType.INTEGER: int,
+    DataType.DATE: int,
+    DataType.STRING: str,
+    DataType.FLOAT: float,
+    DataType.BOOLEAN: bool,
+}
+
+
+class TahomaDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching TaHoma data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        logger: logging.Logger,
+        *,
+        name: str,
+        client: TahomaClient,
+        devices: List[Device],
+        listener_id: str,
+        update_interval: Optional[timedelta] = None,
+    ):
+        """Initialize global data updater."""
+        super().__init__(
+            hass, logger, name=name, update_interval=update_interval,
+        )
+
+        self.client = client
+        self.devices: Dict[str, Device] = {d.deviceurl: d for d in devices}
+        self.listener_id = listener_id
+
+    async def _async_update_data(self) -> Dict[str, Device]:
+        """Fetch data from Tahoma."""
+        events = await self.client.fetch_event_listener(self.listener_id)
```

Comment:
Should there be no try catch here? I see all integrations and samples use `            raise UpdateFailed()`. We can also have exceptions here which we should handle.

Reply:
Done.